src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -0,0 +1,1345 @@
+        // 
+#if !PLATFORM_UNIX
+        public static readonly char VolumeSeparatorChar = ':';
+#else
+        public static readonly char VolumeSeparatorChar = '/';
+#endif // !PLATFORM_UNIX
+
+        // Trim trailing white spaces, tabs etc but don't be aggressive in removing everything that has UnicodeCategory of trailing space.
+        // String.WhitespaceChars will trim aggressively than what the underlying FS does (for ex, NTFS, FAT).    
+        internal static readonly char[] TrimEndChars = { (char)0x9, (char)0xA, (char)0xB, (char)0xC, (char)0xD, (char)0x20, (char)0x85, (char)0xA0 };
+
+#if !PLATFORM_UNIX
+        private static readonly char[] RealInvalidPathChars = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31 };
+
+        // This is used by HasIllegalCharacters
+        private static readonly char[] InvalidPathCharsWithAdditionalChecks = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31, '*', '?' };
+
+        private static readonly char[] InvalidFileNameChars = { '\"', '<', '>', '|', '\0', (Char)1, (Char)2, (Char)3, (Char)4, (Char)5, (Char)6, (Char)7, (Char)8, (Char)9, (Char)10, (Char)11, (Char)12, (Char)13, (Char)14, (Char)15, (Char)16, (Char)17, (Char)18, (Char)19, (Char)20, (Char)21, (Char)22, (Char)23, (Char)24, (Char)25, (Char)26, (Char)27, (Char)28, (Char)29, (Char)30, (Char)31, ':', '*', '?', '\\', '/' };
+#else
+        private static readonly char[] RealInvalidPathChars = { '\0' };
+
+        // This is used by HasIllegalCharacters
+        private static readonly char[] InvalidPathCharsWithAdditionalChecks = { '\0', '*', '?' };
+
+        private static readonly char[] InvalidFileNameChars = { '\0', '*', '?', '\\', '/' };
+#endif // !PLATFORM_UNIX
+
+#if !PLATFORM_UNIX
+        public static readonly char PathSeparator = ';';
+#else
+        public static readonly char PathSeparator = ':';
+#endif // !PLATFORM_UNIX
+
+
+        // Make this public sometime.
+        // The max total path is 260, and the max individual component length is 255. 
+        // For example, D:\<256 char file name> isn't legal, even though it's under 260 chars.
+        internal static readonly int MaxPath = 260;
+        private static readonly int MaxDirectoryLength = 255;
+
+        // Windows API definitions
+        internal const int MAX_PATH = 260;  // From WinDef.h
+        internal const int MAX_DIRECTORY_PATH = 248;   // cannot create directories greater than 248 characters
```

Comment:
These constants are all wrong on Unix.  We already have a work item to fix it in mscorlib, but this adds yet another place to be addressed.  Can you open a corefx work item to fix these, too?